rsmtool/prmse_utils.py

```diff
@@ -0,0 +1,313 @@
+"""
+Utility classes and functions related to computing test
+theory based evaluations
+
+:author: Anastassia Loukina (aloukina@ets.org)
+:author: Jeremy Biggs (jbiggs@ets.org)
+:author: Nitin Madnani (nmadnani@ets.org)
+
+:date: 05/20/2019
+:organization: ETS
+"""
+
+import pandas as pd
+
+
+def compute_variance_of_errors(df,
+                               h1_column='sc1',
+                               h2_column='sc2',):
+    """
+    Compute variance of errors in human scores
+
+    Parameters
+    ----------
+    df : pandas DataFrame
+        Input data frame. Must contain columns `h1_column` and `h2_column`
+    h1_column : str, optional
+        The first human score column name.
+        Defaults to 'sc1'.
+    h2_column : str, optional
+        The second human score column name.
+        Defaults to 'sc2'.
+
+    Returns
+    -------
+    v_e: float
+         Variance of errors in human scores
+
+    Raises
+    ------
+    ValueError
+         If any of the values in `h2_column` or `h1_column` are NaN.
+    """
+
+    # check that all responses are double scored
+    if df[h2_column].isnull().any() or df[h1_column].isnull().any():
+        raise ValueError("Variance of errors should only be computed on double-scored responses.")
+
+    N = len(df)
+
+    v_e = 1 / (2 * N) * ((df[h1_column] - df[h2_column])**2).sum()
+
+    return v_e
+
+
+def compute_true_score_var_subset_double_scored(single_human_scores,
+                                                double_human_scores,
+                                                v_e):
+    """
+    Compute variance of true scores
+    in a situation where only some
+    of the responses are double scored.
+
+    Parameters
+    ----------
+    single_human_scores : pandas Series
+        Human scores for single-scored responses
+    double_human_scores : pandas Series
+       Human scores for double-scored responses
+    v_e : float
+        Variance of errors in human scores
+
+    Returns
+    -------
+    var_t : float
+        Variance of true scores
+    """
+
+    # N of examinees with 1 human score.
+    n_1 = len(single_human_scores)
```

Comment:
I disagree - the formulas later on become unreadable if we use long variable names. Since this is only used within one function and is explained in comments, I think it's ok to have a short name.